### PR TITLE
DAOS-15859 build: Move master to 2.7 test builds.

### PR DESCRIPTION
**Pull request by phender**
Moving master to 2.7 test builds. Bump version to 2.7.100.

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -15,11 +15,11 @@ In addition to on-prem deployments, DAOS can be deployed in cloud
...
 in the [DAOS in the Cloud](./cloud/) section.
 
-The included document versions are associated with DAOS v2.5.
+The included document versions are associated with DAOS v2.7.
 They may also describe features that are currently under development
 for a future DAOS release. Those features will be clearly marked as such.
 
 !!! warning
-    DAOS v2.5 is the (unstable) master branch for DAOS v2.6 development.
+    DAOS v2.7 is the (unstable) master branch for DAOS v2.6 development.
```
Comment (daltonbohning) [inline docs/index.md:23]:
```suggestion
    DAOS v2.7 is the (unstable) master branch for DAOS v2.8 development.
```

File: docs/release/release_notes.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
-# DAOS Version 2.6 Release Notes
+# DAOS Version 2.7 Release Notes
 
-DAOS 2.6 is under active development and has not been released yet.
+DAOS 2.7 is under active development and has not been released yet.
```
Comment (daltonbohning) [inline docs/release/release_notes.md:3]: 2.8 here since 2.7 will not be released

File: docs/release/support_matrix.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
-# DAOS Version 2.6 Support
+# DAOS Version 2.7 Support
 
-DAOS 2.6 is under active development and has not been released yet.
+DAOS 2.7 is under active development and has not been released yet.
```
Comment (daltonbohning) [inline docs/release/support_matrix.md:3]: 2.8

File: docs/release/upgrading.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
-# Upgrading to DAOS Version 2.6
+# Upgrading to DAOS Version 2.7
 
-DAOS 2.6 is under active development and has not been released yet.
+DAOS 2.7 is under active development and has not been released yet.
```
Comment (daltonbohning) [inline docs/release/upgrading.md:3]: 2.8

**Review by daltonbohning — COMMENTED** [review]
I think this needs updating
https://raw.githubusercontent.com/daos-stack/daos/master/docs/admin/administration.md
Under "Examples:"
2.4 -> 2.6
2.5 -> 2.7


The previous older references I'm not sure of. Maybe can handle this separately

File: docs/release/release_notes.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
-# DAOS Version 2.6 Release Notes
+# DAOS Version 2.7 Release Notes
```
Comment (daltonbohning) [inline docs/release/release_notes.md:1]: 2.8

File: docs/release/support_matrix.md [hunk 6]
```diff
@@ -1,6 +1,6 @@
-# DAOS Version 2.6 Support
+# DAOS Version 2.7 Support
```
Comment (daltonbohning) [inline docs/release/support_matrix.md:1]: 2.8

File: docs/release/upgrading.md [hunk 7]
```diff
@@ -1,6 +1,6 @@
-# Upgrading to DAOS Version 2.6
+# Upgrading to DAOS Version 2.7
```
Comment (daltonbohning) [inline docs/release/upgrading.md:1]: 2.8

**Review by daltonbohning — APPROVED** [review]

**Review by mchaarawi — APPROVED** [review]

**Comment by phender** [comment]
All tests passed in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-14407/3/testReport/

**Comment by brianjmurrell** [comment]
FWIW, all of the doc updates that are required seem to be begging for some form of templating.  The numbers of changes such as this is just going to continue to grow and be more troublesome for future updates such as this.

**Comment by daltonbohning** [comment]
> FWIW, all of the doc updates that are required seem to be begging for some form of templating. The numbers of changes such as this is just going to continue to grow and be more troublesome for future updates such as this.

markdown doesn't seem to have any built-in template/variables so we'd either need some plugin (if they exist) or some template + script to update.

We _**always**_ forget one or more references on the first go :(

**Comment by brianjmurrell** [comment]
> markdown doesn't seem to have any built-in template/variables

No, it does not, sadly.

> so we'd either need some plugin (if they exist) or some template + script to update.

Indeed.  The latter is what I was thinking about.  If our project were a [GNU Autotools](https://en.wikipedia.org/wiki/GNU_Autotools) project for example, we could template any file we wanted by creating them as `.in` files with `@…@` variables in them and one of the _Autotools_ will replace the variables with values.

We could do much the same much simpler with `sed`, or even employ a more sophistacated templating engine like [Jinja](https://palletsprojects.com/p/jinja/).
 
> We _**always**_ forget one or more references on the first go :(

Exactly!

**Pull request merged**